CONTRIBUTING.md

```diff
@@ -0,0 +1,324 @@
+
+```
+from transformers import PretrainedConfig
+
+class UFDClassifierConfig(PretrainedConfig):
+
+    model_type = "classifier"
+
+    def __init__(self, embed_dim=1024, num_class=2, initrange=0.1, **kwargs):
+        super().__init__(**kwargs)
+        self.embed_dim = embed_dim
+        self.num_class = num_class
+        self.initrange = initrange
+```
+
+For models that use or adapt pre-trained configs already available in the `transformers` package, the model config should inherit from the pre-trained config class instead. For example, this model config inherits from `BertConfig` which is a child class of `PretrainedConfig`.
+
+```
+from transformers import BertConfig
+
+class NewModelConfig(BertConfig):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+```
+
+### Preprocess
+
+The `preprocess.py` script and its associated preprocessor class is an addition in `sgnlp`. When implementing various models, the team found that more complex models required more complex preprocessing. For example, some NLP models take in multiple different text inputs (ie, different utterances, multiple tweets, a single question and multiple possible answers, etc) which require different preprocessing steps. The `preprocess.py` and the preprocessor class is the team's solution to packaging all of these different steps into a single and consistent (across different models) step.
+
+The preprocessor class inherits from the default `object` class. All preprocessing steps should be executed in the class' `__call__` method. The `__call__` method should return a dictionary containing all the necessary input tensors required by the model. The following code snippet illustrates the `__call__` method from the RECCON Span Extraction model's `RecconSpanExtractionPreprocessor`.
+
+```
+class RecconSpanExtractionPreprocessor:
+
+    def __call__(
+        self, data_batch: Dict[str, List[str]]
+    ) -> Tuple[
+        BatchEncoding,
+        List[Dict[str, Union[int, str]]],
+        List[SquadExample],
+        List[SquadFeatures],
+    ]:
+        self._check_values_len(data_batch)
+        concatenated_batch, evidences = self._concatenate_batch(data_batch)
+        dataset, examples, features = load_examples(
+            concatenated_batch, self.tokenizer, evaluate=True, output_examples=True
+        )
+
+        input_ids = [torch.unsqueeze(instance[0], 0) for instance in dataset]
+        attention_mask = [torch.unsqueeze(instance[1], 0) for instance in dataset]
+        token_type_ids = [torch.unsqueeze(instance[2], 0) for instance in dataset]
+
+        output = {
+            "input_ids": torch.cat(input_ids, axis=0),
+            "attention_mask": torch.cat(attention_mask, axis=0),
+            "token_type_ids": torch.cat(token_type_ids, axis=0),
+        }
+        output = BatchEncoding(output)
```

Comment:
@raymondng76 and I used `BatchEncoding` to create the input tensors, which functions like a dictionary and a class. Should we just return the input tensors as dictionary directly instead of returning it as a BatchEncoding object? This will help to simplify things for contributors when they refer to the existing models as examples.